packages/aws-library/src/aws_library/ec2/models.py

```diff
@@ -84,14 +84,17 @@ class AWSTagValue(ConstrainedStr):
 @dataclass(frozen=True)
 class EC2InstanceData:
     launch_time: datetime.datetime
-    id: str  # noqa: A003
+    id: str
     aws_private_dns: InstancePrivateDNSName
     aws_public_ip: str | None
-    type: InstanceTypeType  # noqa: A003
+    type: InstanceTypeType
     state: InstanceStateNameType
     resources: Resources
     tags: EC2Tags
 
+    def __hash__(self) -> int:
+        return hash(self.id)
```

Comment:
So optimistic: I would bring in all the fields. Nothing blocks you from having 2 instances with the same id. Just `hash(self.__dict__)`